### PR TITLE
chore(docs): add troubleshooting note when installation says Cython is not available

### PR DESCRIPTION
**Pull request by r1viollet**
## Description

When running on old an old debian distribution, I came across this issue. A better fix is to warn the user during installation with an explicit message, though this note could still be useful to users in the mean time.

## Checklist
- [ ] Followed the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines) when writing a release note.
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Motivation

Explain to users why installation is failing.

## Reviewer Checklist
- [x] Title is accurate.
- [x] Description motivates each change.
- [x] No unnecessary changes were introduced in this PR.
- [x] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Tests provided or description of manual testing performed is included in the code or PR.
- [x] Release note has been added and follows the [library release note guidelines](https://ddtrace.readthedocs.io/en/stable/contributing.html#Release-Note-Guidelines), or else `changelog/no-changelog` label added.
- [x] All relevant GitHub issues are correctly linked.
- [x] Backports are identified and tagged with Mergifyio.

### CODE REVIEW

File: docs/troubleshooting.rst [hunk 1]
```diff
@@ -2,6 +2,14 @@
 Troubleshooting
 ===============
 
+Installation failure
+====================
+
+``ModuleNotFoundError: No module named 'Cython'``
+
+pip is failing to install `dd-trace` and complaining about a missing module (Cython).
```
Comment (brettlangdon) [inline docs/troubleshooting.rst:10]:
```suggestion
pip is failing to install ``ddtrace`` and complaining about a missing module (Cython).
```

File: docs/troubleshooting.rst [hunk 2]
```diff
@@ -2,6 +2,14 @@
...
+Ensure that pip is installed with version 18 or above.
```
Comment (brettlangdon) [inline docs/troubleshooting.rst:11]:
```suggestion
``pip>=18`` is required to properly install ``ddtrace`` package.

Check which version of pip you are using with ``pip --version``.

Consider upgrading pip via ``pip install -U pip>=18``.
```

**Review by brettlangdon — COMMENTED** [review]
just some minor suggestions, otherwise, thanks!

File: docs/troubleshooting.rst (hunk 1 above)
Comment (r1viollet) [reply docs/troubleshooting.rst]: My bad, thanks for reviewing this and submitting improvements.

File: docs/troubleshooting.rst [hunk 3]
```diff
@@ -2,6 +2,18 @@
 Troubleshooting
 ===============
 
+Installation failure
+====================
+
+``ModuleNotFoundError: No module named 'Cython'``
+
+pip is failing to install `ddtrace` and complaining about a missing module (Cython).
```
Comment (brettlangdon) [inline docs/troubleshooting.rst:10]:
```suggestion
pip is failing to install ``ddtrace`` and complaining about a missing module (Cython).
```

this is rST not markdown, so we need double back ticks

**Review by brettlangdon — APPROVED** [review]

**Review by gnufede — APPROVED** [review]

**Comment by r1viollet** [comment]
Thanks for following up and merging :bow:

**Pull request merged**